assembly/managed/held.ts

```diff
@@ -115,9 +144,26 @@ export class RequestHeldEvent<T, U, UReturn> extends HeldEvent<T> {
     super();
   }
 
-  handle(ctx: HeldContext<T>, msg: Message<HeldEvent<T>>): bool {
-    let value = this.callback(this.value, ctx);
-    msg.reply<UReturn>(value);
+  handle(ctx: HeldContext<T>, msg: Message<HeldEvent<T>>, timeout: u64 = u64.MAX_VALUE): bool {
+    let sandboxCtx = new RequestHeldEventSandboxContext(this, ctx);
+    let result = sandbox<
+      RequestHeldEventSandboxContext<T, U, UReturn>,
+      RequestHeldEventSandboxReturn<T, UReturn>
+    >(
+      sandboxCtx,
+      (val: RequestHeldEventSandboxContext<T, U, UReturn>) => {
+        let ret = val.event.callback(val.event.value, val.ctx);
+        return new RequestHeldEventSandboxReturn<T, UReturn>(val.ctx, ret);
+      },
+      timeout
+    );
+
+    if (result.isOk()) {
+      let unboxed = result.expect().value;
+      ctx.value = unboxed.ctx.value;
+      msg.reply<UReturn>(unboxed.ret);
```

Comment:
What if the result isn't okay? What would happen if there wasn't any reply?

Reply:
The request would have timed out... But I made it more explicit.